src/types.hpp

```diff
@@ -353,10 +373,31 @@ struct AgentID {
         float id;
     };
 
-    const size_t AbsoluteSelfObservationExportSize = 13; // 3 + 4 + 1 + 2 + 2
+    const size_t AbsoluteSelfObservationExportSize = 14; // 3 + 5 + 2 + 3 + 1
 
     static_assert(sizeof(AbsoluteSelfObservation) == sizeof(float) * AbsoluteSelfObservationExportSize);
 
+    //Metadata struct : using agent IDs.
+    struct MetaData
+    {
+        int32_t id;
```

Comment:
Does MetaData need id field? We already have an AgentId struct and we export it in the selfobservation. I think this field is redundant.

Reply:
I think its better to move id into the agent metadata, since getting the id at a particular flag value is easy this way. I think the question is whether id is better off in selfobs or metadata, I am inclined toward the latter

Reply:
I disagree and argue for the former. The reason is that we want to directly access the id of an object in sim.cpp. I think the term "metadata" implies optional data about objects while id is an integral part of the agent. This also follows the convention for road objects where they have a RoadId. 

Also, this feature of agent id was requested by Hyeonchang who needs the id field as part of his RL training. I think selfobservation gives identifying information about an agent, and we should not need to query the metadata to find out the id of an agent.

Reply:
Hmm I see what you mean. My understanding was that observation implies some view of the state, and so the id shouldn't be in it. Regardless I do think metadata should have IDs, else we'd have to index the selfobs with the metadata to get a particular id.